src/govuk/components/summary-list/_index.scss

```diff
@@ -109,37 +110,25 @@
 
   // No border on entire summary list
   .govuk-summary-list--no-border {
-    @include govuk-media-query($until: tablet) {
-      .govuk-summary-list__row {
-        border: 0;
-      }
+    .govuk-summary-list__row {
+      border: 0;
     }
 
-    @include govuk-media-query($from: tablet) {
-      .govuk-summary-list__key,
-      .govuk-summary-list__value,
-      .govuk-summary-list__actions {
-        // Remove 1px border, add 1px height back on
-        padding-bottom: govuk-spacing(2) + 1px;
-        border: 0;
-      }
+    .govuk-summary-list__key,
+    .govuk-summary-list__value,
+    .govuk-summary-list__actions {
+      padding-bottom: govuk-spacing(2) + 1px;
```

Comment:
I don't know if this is really needed, but opted to maintain the existing design and behaviour rather than introduce changes.